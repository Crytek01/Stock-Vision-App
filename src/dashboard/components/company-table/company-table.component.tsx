import { FunctionComponent, useMemo } from "react";
import {
  DataGrid,
  GridColDef,
  GridPaginationModel,
  GridSortDirection,
  GridSortModel,
} from "@mui/x-data-grid";
import { TStockVisionCompaniesInfo } from "@/pages/api/stocks/company";
import { Button } from "@mui/material";

interface ICompanyTableProps {
  data: TStockVisionCompaniesInfo;
  onRowDelete?: (rowId: string) => void;
}

const createCompanyTableColumns = (
  onRowDelete?: (rowId: string) => void
): GridColDef[] => {
  const columns: GridColDef[] = [
    { field: "companyName", headerName: "Company Name", flex: 1 },
    { field: "tickerSymbol", headerName: "Ticker Symbol", flex: 1 },
    { field: "listedSince", headerName: "Listed Since", flex: 1, type: "date" },
    {
      field: "marketCapitalization",
      headerName: "Market Cap",
      flex: 1,
      type: "number",
    },
    {
      field: "paysDividends",
      headerName: "Pays Dividends",
      flex: 1,
      type: "boolean",
    },
    { field: "stockPrice", headerName: "Stock Price", flex: 1, type: "number" },
  ];

  if (onRowDelete) {
    const deleteRow: GridColDef = {
      field: "delete",
      headerName: "",
      width: 90,
      renderCell: (params) => (
        <Button
          variant="outlined"
          color="secondary"
          onClick={() => onRowDelete(params.id as string)}
        >
          Delete
        </Button>
      ),
    };
    columns.push(deleteRow);
  }

  return columns;
};

const sortingOrder: GridSortDirection[] = ["asc", "desc"];
const sortingModel: GridSortModel = [
  {
    field: "stockPrice",
    sort: "asc",
  },
];

const gridPaginationModel: GridPaginationModel = {
  page: 0,
  pageSize: 8,
};

const pageSizeOptions: number[] = [8];

export const CompanyTable: FunctionComponent<ICompanyTableProps> = ({
  data = [],
  onRowDelete,
}) => {
  const columns = useMemo(() => {
    return createCompanyTableColumns(onRowDelete);
  }, [onRowDelete]);

  return (
    <DataGrid
      rows={data}
      columns={columns}
      paginationModel={gridPaginationModel}
      sortingMode="client"
      sortingOrder={sortingOrder}
      rowSelection={true}
      sortModel={sortingModel}
      pageSizeOptions={pageSizeOptions}
    />
  );
};
