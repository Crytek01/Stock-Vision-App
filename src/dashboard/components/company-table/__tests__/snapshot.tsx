import { render } from "@testing-library/react";
import { CompanyTable } from "../company-table.component";

it("renders CompanyTable component", () => {
  const { container } = render(<CompanyTable data={[]} />);
  expect(container).toMatchSnapshot();
});
