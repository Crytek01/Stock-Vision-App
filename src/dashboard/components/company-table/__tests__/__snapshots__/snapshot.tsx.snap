// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CompanyTable component 1`] = `
<div>
  <div
    aria-colcount="6"
    aria-multiselectable="false"
    aria-rowcount="1"
    class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1usso8h-MuiDataGrid-root"
    role="grid"
  >
    <div
      class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
    >
      <div
        class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
        style="min-height: 56px; max-height: 56px; line-height: 56px;"
      >
        <div
          class="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
          role="rowgroup"
        />
      </div>
    </div>
    <div
      class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
    >
      <div />
      <div
        class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            0â€“0 of 0
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
